.github/workflows/testing-europe-deploy-arena-release.yml

```
@@ -0,0 +1,77 @@
+name: Deploy to Europe testing
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        type: choice
+        description: What release to deploy
+        options:
+          - game_backend
+          - arena
+          - arena_load_test
+          - game_client
+        required: true
+
+jobs:
+  build-deploy:
+    name: Build and deploy to Europe testing
+    runs-on: ubuntu-latest
+    environment:
+      name: testing
+      url: https://europe-testing.curseofmirra.com/
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Tailscale
+        uses: tailscale/github-action@v2
+        with:
+          oauth-client-id: ${{ secrets.TS_OAUTH_CLIENT_ID }}
+          oauth-secret: ${{ secrets.TS_OAUTH_SECRET }}
+          tags: tag:ci
+
+      - name: Create ssh private key file from env var
+        env:
+          SSH_KEY: ${{ secrets.SSH_KEY }}
+          TS_HOST: ${{ vars.TS_EUROPE_HOST }}
+        run: |
+          set -ex
+          mkdir -p ~/.ssh/
+          sed -E 's/(-+(BEGIN|END) OPENSSH PRIVATE KEY-+) *| +/\1\n/g' <<< "$SSH_KEY" > ~/.ssh/id_ed25519
+          chmod 400 ~/.ssh/id_ed25519
+          retries=5; until ssh-keyscan $TS_HOST >> ~/.ssh/known_hosts || [ $retries -eq 0 ]; do ((retries--)); sleep 5; done
+
+      - name: Copy deploy script
+        env:
+          SSH_USERNAME: ${{ secrets.SSH_USERNAME }}
+          SSH_HOST: ${{ vars.TS_EUROPE_HOST }}
+        run: |
+          set -ex
+          rsync -avz --mkpath /devops/deploy.sh ${SSH_USERNAME}@${SSH_HOST}:/home/${SSH_USERNAME}/deploy-script/
```

```suggestion
          rsync -avz --mkpath devops/deploy.sh ${SSH_USERNAME}@${SSH_HOST}:/home/${SSH_USERNAME}/deploy-script/
```
the `deploy.sh` path shall not be absolute!

---

Devops sight ❇️